src/components/subscriptions/SubscriptionExpirationModal.jsx

```diff
@@ -0,0 +1,137 @@
+        Your current subscription is set to expire in {daysUntilExpiration} days.
+        In order to minimize course access disruptions for your learners, make sure your invoice has
+        been completed.
+      </p>
+      <p>
+        If you have questions or need help, please contact the edX Customer Success team at
+        {' '}
+        <MailtoLink to="customersuccess@edx.org">customersuccess@edx.org</MailtoLink>.
+      </p>
+      <i>
+        Access expires on {formatTimestamp({ timestamp: expirationDate })}
+      </i>
+    </React.Fragment>
+  );
+
+  const renderExpiredBody = () => (
+    <React.Fragment>
+      <Img className="w-25 my-5 mx-auto d-block" src={edxLogo} alt="edX logo" />
+      <p>
+        Your subscription license expired on {formatTimestamp({ timestamp: expirationDate })}.
+        To access your subscription management page contact edX and reactivate your subscriptions.
+      </p>
+      <p>
+        What to do next?
+      </p>
+      <ul>
+        <li>
+          To reactivate your subscriptions please cotact the edX Customer Success team at
+          {' '}
+          <MailtoLink to="customersuccess@edx.org">customersuccess@edx.org</MailtoLink>
+        </li>
+        <li>
+          View your learner progress in the <Link to={`/${enterpriseSlug}/admin/learners`}>learner management page</Link>
+        </li>
+        <li>
+          Manage your codes in the <Link to={`/${enterpriseSlug}/admin/coupons`}>code management page</Link>
+        </li>
+      </ul>
+    </React.Fragment>
+  );
+
+  // If the subscription has already expired, we show a different un-dismissible modal
+  const subscriptionExpired = daysUntilExpiration <= 0;
+  if (subscriptionExpired) {
+    return (
+      <Modal
+        renderHeaderCloseButton={false}
+        renderDefaultCloseButton={false}
+        title={null}
+        body={renderExpiredBody()}
+        open
+      />
+    );
+  }
+
+  if (daysUntilExpiration > SUBSCRIPTION_EXPIRATION_FIRST_THRESHOLD) {
+    return null;
+  }
+
+  let expirationThreshold = SUBSCRIPTION_EXPIRATION_FIRST_THRESHOLD;
+  if (details.daysUntilExpiration <= SUBSCRIPTION_EXPIRATION_SECOND_THRESHOLD) {
+    expirationThreshold = SUBSCRIPTION_EXPIRATION_SECOND_THRESHOLD;
+  }
+  if (daysUntilExpiration <= SUBSCRIPTION_EXPIRATION_THIRD_THRESHOLD) {
+    expirationThreshold = SUBSCRIPTION_EXPIRATION_THIRD_THRESHOLD;
+  }
+  const seenCurrentExpirationModalCookieName = `${SEEN_SUBSCRIPTION_EXPIRATION_MODAL_COOKIE_PREFIX}${expirationThreshold}`;
+  const cookies = new Cookies();
+  const seenCurrentExpirationModal = cookies.get(seenCurrentExpirationModalCookieName);
+  // If they have already seen the expiration modal for their current threshold (as deterrmined by
+  // the cookie), don't show them anything
+  if (seenCurrentExpirationModal) {
+    return null;
+  }
+
+  return (
+    <Modal
+      renderHeaderCloseButton={false}
+      title={renderTitle()}
+      body={renderBody()}
+      closeText="Ok"
```

Comment:
The current Paragon modal, as you probably realized, doesn't allow modifying the class name of the default close button unfortunately 😞 

That said, since this is the only action in the modal, it should have the primary variant, matching what's in the wireframe. We might get around this for now by hiding the default close button, and passing our own "Ok" button as a prop, managing whether the modal is open via the Modal's `open` prop:

```jsx
const [isOpen, setIsOpen] = useState(false);

<Modal
  buttons={[
    <Button
      onClick={() => {
        cookies.set(...);
        setIsOpen(false);
      }
    >
      Ok
    </Button>
  ]}
  open={isOpen}
/>
```

Or, alternatively, we could also override the CSS of the default close button to use the `.btn-primary` styles.

Reply:
ah right I missed that. I think just overriding the close button to match `btn-primary` makes sense. Is there a way to get the styles from paragon's scss like bootstrap does here: https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/_buttons.scss#L71-L73? If not I can copy the styles manually